torchgeo/datasets/geo.py

```diff
@@ -635,9 +670,15 @@ def __init__(
                 entry and returns a transformed version
             label_name: name of the dataset property that has the label to be
                 rasterized into the mask
+            task: what is the tas the dataset is used for. Supported output types
+               `object_detection`, `semantic_segmentation`, `instance_segmentation`
+            gpkg_layer: if the input is a multilayer geopackage, specify which layer
```

Comment:
I believe this can work with files other than just gpkg so let's just call this arg "layer"